django/contrib/admin/options.py

```diff
@@ -1590,9 +1608,12 @@ def _changeform_view(self, request, object_id, form_url, extra_context):
                 form_validated = False
         else:
             if add:
-                initial = self.get_changeform_initial_data(request)
+                initial, formset_initial_data = self.get_changeform_initial_data(request)
                 form = ModelForm(initial=initial)
                 formsets, inline_instances = self._create_formsets(request, form.instance, change=False)
+                # setting the initial value of the formset
+                for formset_index in range(len(formsets)):
+                    formsets[formset_index].initial = formset_initial_data[formset_index]
```

Comment:
I don't think the comment is necessary - it seems obvious what is happening.

You can also dispense with the index-based looping:
```python
for formset, data in zip(formsets, formset_initial_data):
    formset.initial = data
```

There is also an assumption that there is an entry in `formset_initial_data` for every formset. I guess that works because of the use of `defaultdict`.